anesthetic/samples.py

```diff
@@ -244,8 +244,12 @@ def plot_1d(self, axes=None, *args, **kwargs):
         for x, ax in axes.items():
             if x in self and kwargs['kind'] is not None:
                 xlabel = self.get_label(x)
-                self[x].plot(ax=ax, xlabel=xlabel,
-                             *args, **kwargs)
+                if x == 'logL_birth':
+                    self[x].replace(-np.inf, np.nan
+                                    ).dropna().plot(ax=ax, xlabel=xlabel,
+                                                    *args, **kwargs)
+                else:
+                    self[x].plot(ax=ax, xlabel=xlabel, *args, **kwargs)
```

Comment:
* The `.dropna()` shouldn't be necessary.
* Shall we separate inf replacement and plotting to reduce code duplication? We could have 
  ```python
  if x == 'logL_birth':
      selfx = self[x].replace(...)
  else:
      selfx = self[x]
  selfx.plot(...)
  ```
* I feel like changing infs to nans shouldn't happen silently. Shall we at least have a warning that we are setting `-inf` to `nan` for plotting purposes?

Reply:
dropna is necessary for the 2D KDE, but not otherwiseo so I've removed those.
I've reduced code repetition as suggested.
With regard to infs to nans, I'm not that keen to have a warning for default behaviour (i.e. `samples.plot_1d()`)

Reply:
> With regard to infs to nans, I'm not that keen to have a warning for default behaviour (i.e. `samples.plot_1d()`)

Well, I'd argue here that `samples.plot_1d()` or `samples.plot_2d()` is not default behaviour for nested sampling data (nor for anesthetic dataframes in general, which typically come with very many columns), otherwise this would have come up earlier, too. By default the user should specify which columns are to be plotted. 

So I think a warning message could help the new user (who is most likely to call the plotting functions without specifying columns) to learn faster how to use the plotting command. This is also how this PR came to be: new user simply trying `samples.plot_1d()` to see if it works...

I agree that too many warning messages are annoying and get in the way, but how often do we intentionally plot the posterior distribution of `logL_birth`? Also, for people that indeed want to intentionally plot `logL_birth`, they can easily avoid the warning by dropping infs first.

Reply:
I am persuaded by this, so think we should have a warning message.

If we're going down a warning route, should we restrict this to logL_birth ? The code would be much neater if we just moved the relevant inf guards into `kde_contour_plot_2d` and `kde_1d` and `kde_2d`. This would then be more consistent with the other functions behaviour, e.g. `samples.plot()`, which just ignores nans and infs.

Reply:
My input from stumbling across this discussion via #332 

Perhaps the default behavior is sensible (include nlive logl and loglbirth in the plot if nothing is specified), but it strikes me that the easiest thing to make the various kde errors less impactful is add a kind=scatter and make that the default for the specific case that the columns to plot are empty?